subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskInputFilePropertiesIntegrationTest.groovy

```diff
@@ -157,29 +157,6 @@ class TaskInputFilePropertiesIntegrationTest extends AbstractIntegrationSpec imp
         executed ":foo"
     }
 
-    @Issue("https://github.com/gradle/gradle/issues/9674")
-    def "allows @Input of task with no actions to be null"() {
-        buildFile << """
-            class FooTask extends DefaultTask {
-               @Input
```

Comment:
Let's replace `@Input` with `@InputFiles` instead. That should expose the problem of https://github.com/gradle/gradle/issues/9674 without running into a validation problem.